src/context-providers/builder/hook.ts

```diff
@@ -1,20 +1,45 @@
-import { useContext } from "react";
+import { useCallback, useContext } from "react";
 import { BuilderContext } from "./provider";
 
 export const useBuilder = () => {
     const { state, dispatch } = useContext(BuilderContext);
 
-    const togglePanel = (isCollapsed: boolean) => {
+    const togglePanel = useCallback((isCollapsed: boolean) => {
         dispatch({
             type: "toggle-panel",
             payload: isCollapsed,
         });
-    };
+    }, []);
+
+    const addElement = useCallback((elementId: string) => {
+        const newElementIds = [...state.elementIds, elementId];
+        dispatch({
+            type: "update-element-ids",
+            payload: newElementIds,
+        });
+    }, []);
```

Comment:
need to add `state.elementIds` as dependency

Reply:
Thanks for pointing! Missed it